client/services/CodeStorageService.js

```diff
@@ -0,0 +1,42 @@
+// Copyright 2017 The TIE Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/*
+ * A service that saves student code to the browser's localStorage.
+ */
+tie.factory('CodeStorageService', [
+  function() {
+    var codeStorageService = {};
+    
+    codeStorageService.getLocalStorageHashKey = function(questionId, language) {
+      return "[" + questionId + ":" + language + "]";
```

Comment:
Would be also nice to add karma tests in assets/tests/QuestionSchemaValidationServiceSpec.js to force the question id to contain only alphanumeric chars + digits (and, in particular, no colons). If we ever want to decode keys this might help with preventing ambiguity in the decoding.

Feel free to do that in this PR (it's quite straightforward) or file a separate issue, but we should make sure this gets tracked and done. Thanks!